### PR TITLE
Add TestPodTainsTolerationIdentifier to check if pod

### PR DESCRIPTION
**Pull request by hamadise**
has any tolerations.
Add UT for the test function

Release Note:
Add TestPodTainsTolerationIdentifier

### CODE REVIEW

File: cnf-certification-test/lifecycle/suite.go [hunk 1]
```diff
@@ -57,6 +58,7 @@ var _ = ginkgo.Describe(common.LifecycleTestKey, func() {
 	testContainersLivenessProbe(&env)
 	testPodsOwnerReference(&env)
 	testHighAvailability(&env)
+	testPodTainsToleration(&env)
```
Comment (sebrandon1) [inline cnf-certification-test/lifecycle/suite.go:61]:
```suggestion
	testPodTaintsToleration(&env)
```

**Review by sebrandon1 — CHANGES_REQUESTED** [review]
One typo

File: cnf-certification-test/lifecycle/suite.go [hunk 2]
```diff
@@ -85,13 +87,44 @@ var _ = ginkgo.Describe(common.LifecycleTestKey, func() {
...
 
+func testPodTaintsTolerationHelper(put *v1.Pod) (bool, error) {
+	if put == nil {
+		return false, errors.New("invalid input, pod under test is nil")
+	}
+	for _, t := range put.Spec.Tolerations {
+		if t.Effect == v1.TaintEffectNoSchedule || t.Effect == v1.TaintEffectPreferNoSchedule ||
+			t.Effect == v1.TaintEffectNoExecute {
+			logrus.Error(fmt.Sprintf("pod %s:%s has toleration %s\n", put.Namespace, put.Name, t.Effect))
+			return false, nil
```
Comment (greyerof) [inline cnf-certification-test/lifecycle/suite.go:98]: You're returning on the first bad toleration found. Unless they're all mutually exclusive, I believe you should iterate through all of them, so you can show all the "bad" ones in the claim file so the customer can fix them altogether. I suggest to change the function to something like:
```
getBadTolerationsTaintEffectFromPod(pod *v1.Pod) []string {
    badEffects := []string{}
    for _, t := range put.Spec.Tolerations {
          if t.Effect == v1.TaintEffectNoSchedule || t.Effect == v1.TaintEffectPreferNoSchedule ||
	          t.Effect == v1.TaintEffectNoExecute {
	          badEffects = append(badEffects , t.Effect)
          }
    }
    return badEffects 
}
```
Then, in the pods loops, you can use check that the output slice is empty:
```
if badEffects := getBadTaintTolerationsEffectsFromPod(put); len(badEffects) > 0 {
    badPods = append...
    tnf.ClaimFilePrintf("%s has invalid taint tolerations effects %v", provider.PodToString(put), badEffects)
}
```

**Review by greyerof — COMMENTED** [review]
You can use provider.PodToString(put) helper function whenever you want to add the pod's name and namespace to a trace or similar.

File: cnf-certification-test/lifecycle/suite.go (hunk 2 above)
Comment (hamadise) [reply cnf-certification-test/lifecycle/suite.go:98]: will fix it

**Pull request closed without merging**